stacker/blueprints/base.py

```diff
@@ -402,6 +406,44 @@ def render_template(self):
         version = hashlib.md5(rendered).hexdigest()[:8]
         return (version, rendered)
 
+    def parse_user_data(self, raw_user_data):
+        """Translate a userdata file to into the file contents.
+
+        It supports referencing template variables to create userdata
+        that's supplemented with information from the data, as commonly
+        required when creating EC2 userdata files. Automatically, encodes
+        the data file to base64 after it is processed.
+
+         Args:
+            raw_userdata (str of userdata):
+                str of the given userdata to be parsed
+
+        """
+        pattern = re.compile(r'{{([::|\w]+)}}')
+        user_data = []
+        start_index = 0
+        variables = self.get_variables()
+        parameters = self.get_cfn_parameters()
+
+        for match in pattern.finditer(raw_user_data):
+            user_data.append(raw_user_data[start_index:match.start()])
+
+            key = match.group(1)
+
+            if key in parameters:
+                user_data.append(parameters[key])
+            elif key in variables:
+                user_data.append(variables[key])
+            else:
+                raise MissingVariable(self.name, key)
+
+            start_index = match.end()
+
+        user_data.append(raw_user_data[start_index:])
+        res = Join("", user_data)
```

Comment:
Since this is just a template, and you're not doing anything w/ Ref, etc, you probably won't need any of this join logic - it'll just be a template.